app/src/main/java/net/gsantner/markor/model/Document.java

```diff
@@ -214,41 +212,46 @@ public static Document fromArguments(Context context, Bundle arguments) {
         return document;
     }
 
-    public synchronized String loadContent(final Context context) {
+    public String loadContent(final Context context) {
+        return loadContent(this, context);
+    }
+
+    public static synchronized String loadContent(final Document document, final Context context) {
 
         String content;
         final char[] pw;
-        if (isEncrypted() && (pw = getPasswordWithWarning(context)) != null) {
+        final File file = document.getFile();
+        if (document.isEncrypted() && (pw = getPasswordWithWarning(context)) != null) {
             try {
-                final byte[] encryptedContext = FileUtils.readCloseStreamWithSize(new FileInputStream(getFile()), (int) getFile().length());
+                final byte[] encryptedContext = FileUtils.readCloseStreamWithSize(new FileInputStream(file), (int) file.length());
                 if (encryptedContext.length > JavaPasswordbasedCryption.Version.NAME_LENGTH) {
                     content = JavaPasswordbasedCryption.getDecryptedText(encryptedContext, pw);
                 } else {
                     content = new String(encryptedContext, StandardCharsets.UTF_8);
                 }
             } catch (FileNotFoundException e) {
-                Log.e(Document.class.getName(), "loadDocument:  File " + getFile() + " not found.");
-                content = "";
+                Log.e(Document.class.getName(), "loadDocument:  File " + file + " not found.");
+                content = null;
             } catch (JavaPasswordbasedCryption.EncryptionFailedException | IllegalArgumentException e) {
                 Toast.makeText(context, R.string.could_not_decrypt_file_content_wrong_password_or_is_the_file_maybe_not_encrypted, Toast.LENGTH_LONG).show();
-                Log.e(Document.class.getName(), "loadDocument:  decrypt failed for File " + getFile() + ". " + e.getMessage(), e);
-                content = "";
+                Log.e(Document.class.getName(), "loadDocument:  decrypt failed for File " + file + ". " + e.getMessage(), e);
+                content = null;
```

Comment:
error = return null

Reply:
maybe

```
catch (encryriptonexception) {
}
catch (exception) {
}
```

to catch all errors at this point. the error-or-not info is contained as null